src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -708,14 +786,24 @@ private IVariableInformation[] ExpandVisualized(IVariableInformation variable)
                         {
                             string processedExpr = ReplaceNamesInExpression(v.Value, variable, visualizer.ScopedNames);
                             Dictionary<string, string> indexDic = new Dictionary<string, string>();
-                            for (uint index = 0; index < size; ++index)
+                            uint currentIndex = (variable as VisualizerWrapper).StartIndex;
```

Comment:
The assumption that variable will always be a `VisualizerWrapper` is causing the test to fail because that is not true with the `showDisplayString` case. You will need to handle where variable is only a `VariableInformation` from the engine.

![image](https://user-images.githubusercontent.com/3953714/184026337-03a23a90-2304-4e48-bdcd-b8a0b9f70d56.png)